sky/provision/azure/instance.py

```diff
@@ -42,6 +44,10 @@
 _POLL_INTERVAL = 1
 
 
+def _use_az_ml() -> bool:
+    return skypilot_config.get_nested(('azure', 'use_az_ml'), False)
```

Comment:
Instead of checking the config in skypilot config, it would be better to keep the config in cluster yaml (e.g., `provider_config` section) and create instances purely based on the cluster yaml. In this case, it is easier to allow old clusters to still work when user change skypiot config.